### ISSUE TITLE
**Bug in `isStringLit()` function**

### ISSUE THREAD

**Issue by pabloosabaterr** [Blopaa/Compiler#1]
**Bug in `isStringLit()` function**

The function currently reports errors when it should only validate string literals, similar to existing functions like `isValidFloatLit()`.

**Current issue:** 
- Function both validates AND reports missing quote errors
- Mixing validation logic with error handling
- Inconsistent with other validation functions (e.g., `isValidFloat()`)

**Proposed fix:**
- Make it a pure validation function like `isValidFloat()`
- Return only true/false without error reporting
- Let the checker handle error reporting separately
- Maintain consistency with existing validation functions

Simple fix for the function's single responsibility principle.

```C
//checks if its a string
int isStringLit(char *val) {
    // ERROR REP SHOULDN'T BE HERE, BUT AS IF SOMETHING HAS QUOTES CANT BE A NUMBER NOR A VARIABLE IT WILL BE TAKEN AS A TRY TO INPUT AN STRING
    // FIX: checker takes anything with a quote and then verifies if it's correct.
    if ((strchr(val, '"') && val[0] != '"') || (strchr(val, '"') && val[strlen(val) - 1] != '"')) {
        repError(ERROR_MISSING_QUOTE, val);
        return 0;
    }
    return val != NULL && val[0] == '"' && val[strlen(val) - 1] == '"';
}```
